internal/proxy/task_index.go

```diff
@@ -144,36 +144,29 @@ func (cit *createIndexTask) parseIndexParams() error {
 			indexParamsMap[kv.Key] = kv.Value
 		}
 	}
-	if !isVecIndex {
-		specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
-		if cit.fieldSchema.DataType == schemapb.DataType_VarChar {
-			if !exist {
-				indexParamsMap[common.IndexTypeKey] = DefaultStringIndexType
-			}
 
-			if exist && !validateStringIndexType(specifyIndexType) {
-				return merr.WrapErrParameterInvalid(DefaultStringIndexType, specifyIndexType, "index type not match")
-			}
-		} else if typeutil.IsArithmetic(cit.fieldSchema.DataType) {
-			if !exist {
-				indexParamsMap[common.IndexTypeKey] = DefaultArithmeticIndexType
-			}
+	specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
+	if exist && specifyIndexType != "" {
+		_, err := indexparamcheck.GetIndexCheckerMgrInstance().GetChecker(specifyIndexType)
+		if err != nil {
+			log.Warn("Failed to get index checker", zap.String(common.IndexTypeKey, specifyIndexType))
+			return merr.WrapErrParameterInvalid("valid index", fmt.Sprintf("invalid index type: %s", specifyIndexType))
+		}
+	}
 
-			if exist && !validateArithmeticIndexType(specifyIndexType) {
-				return merr.WrapErrParameterInvalid(DefaultArithmeticIndexType, specifyIndexType, "index type not match")
-			}
-		} else if typeutil.IsBoolType(cit.fieldSchema.DataType) {
-			if !exist {
-				return merr.WrapErrParameterInvalidMsg("no index type specified")
-			}
-			if specifyIndexType != InvertedIndexType {
-				return merr.WrapErrParameterInvalidMsg("index type (%s) not supported for boolean, supported: %s",
-					specifyIndexType, InvertedIndexType)
+	if !isVecIndex {
+		specifyIndexType, exist := indexParamsMap[common.IndexTypeKey]
+		if Params.AutoIndexConfig.ScalarAutoIndexEnable.GetAsBool() || specifyIndexType == AutoIndexName || !exist {
+			if typeutil.IsArithmetic(cit.fieldSchema.DataType) {
+				indexParamsMap[common.IndexTypeKey] = Params.AutoIndexConfig.ScalarNumericIndexType.GetValue()
+			} else if typeutil.IsStringType(cit.fieldSchema.DataType) {
+				indexParamsMap[common.IndexTypeKey] = Params.AutoIndexConfig.ScalarVarcharIndexType.GetValue()
+			} else if typeutil.IsBoolType(cit.fieldSchema.DataType) {
+				indexParamsMap[common.IndexTypeKey] = Params.AutoIndexConfig.ScalarBoolIndexType.GetValue()
+			} else {
+				return merr.WrapErrParameterInvalid("supported field",
```

Comment:
This error message is not clear, will mislead users to think the data type can't be indexed. Maybe we want to let users to know we don't support auto index for this data type?